### DISCUSSION TITLE
Goodwe ES Gen2 (ET family) support

### DISCUSSION THREAD

**Discussion by DezR** [ai-republic/bms-to-inverter#19] [Q&A] [answered]
Hey ai-republic, thanks for developing this. 
I've been trying to get a Goodwe GW5000-ES-20 hybrid inverter and battery bank with parallel JK BMS's to work together with several other emulators (based on ESP32's, Arduino's, different CAN adapters etc).....but with no success so far.

The ES G2 range is supposedly compatible with a large range of battery types per this [Goodwe approved battery statement](https://www.goodwe.com.au/Ftp/EN/Downloads/User%20Manual/GW_Approved%20Battery%20Options%20Statement-EN.pdf) , however will not permit any battery charging or discharging no matter what emulated battery type is chosen or how the inverter is configured.

From speaking with Goodwe, they've said that the onus is on the battery OEM to make their BMS compatible and aren't really sure if every battery on the approved battery list will actually work.

Goodwe also told me that they no longer support the unspecified battery profiles (like Self-Defined, Default Lithium or Lead Acid) that are still selectable in their Solar Go or PV Master app. I can confirm that these don't work with or without BMS or CAN connection.

It seems that the newest G2 range uses a different CAN scheme that doesn't work with the battery emulators that are already out there, yet still supposedly and somehow works with the Goodwe approved battery types.

Have you come across this type of problem with your emulator/translator and could it be made to work with the latest Goodwe inverters?

**Comment by ai-republic** [comment]
Hi,
I have a Goodwe HV CAN specification which works for GW6000-EH here, but have not yet implemented a binding.
For Goodwe LV inverters CAN specification I haven't found it.

According to their battery statement the ES should be able to communication via the Pylon LV CAN protocol with their appropriate firmware installed.

So it should be doable to communicate to your inverter via the Pylon or the BYD CAN protocol.

**Comment by ai-republic** [comment]
I've just added a BYD BMS for LV CAN binding for you

**Comment by DezR** [comment]
Thank you for your effort and help.
My PI5 arrives later today and will try this.

I've also unsuccessfully been searching for the Goodwe CAN spec and only found a MODBUS doc from 2017. 
I'll try to contact Goodwe in Suzhou to see what they can help with.

**Comment by DezR** [comment, marked as answer]
As an update for anyone with the same (or ET family) inverter, Goodwe (finally) provided a firmware update a few weeks back that permits 'Self Define' mode to be selected and actually function. 
It's limited to 45A charge/discharge and doesn't 'communicate' with any DIY batteries that use JK, JBD or Daly BMS's.
This where this BMS-to-inverter emulator is so important :)

**Reply by ai-republic** [reply]
Well, it's actually not an emulator - its a conversion bridge to read data from the BMS in its communication protocol and convert it to a communication protocal understood by the inverter :)

**Reply by DezR** [reply]
Ha. Fair point :)
The big advantage with your work is that multiple BMS's ( and potentially smart balancers and shunts) can be aggregated without additional hardware.

The potential is there for LV or HV and better M&C (than Arduino or ESP32).
